apis/duck/v1/knative_reference.go

```diff
@@ -42,6 +47,11 @@ type KReference struct {
 
 	// API version of the referent.
 	APIVersion string `json:"apiVersion"`
+
```

Comment:
I think we need to update L49 to be omitempty & +optional?

Reply:
TBH I don't know what `+optional` is useful for :smile:

Reply:
Tried to run the codegen and nothing changes using `+optional`

Reply:
FYI:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

And in particular:
```
Note that for backward compatibility, any field that has the omitempty struct tag will be considered to be optional, but this may change in the future and having the +optional comment tag is highly recommended.
```

Reply:
ok cool, adding it